### ISSUE TITLE
Only the front USB port is working

### ISSUE THREAD

**Issue by henryxrl** [SchmockLord/Hackintosh-Intel-i9-10900k-AsRock-Z490-Phantom-ITX-TB3#56]
I have the same CPU and mobo as yours with RX 6800XT GPU. So your EFI folder should just work.
Everything else works flawlessly except for USB ports (Thunderbolt port works). None of the USB ports on the back of the mobo works - only the USB of the front I/O works. Could you help me with this?

![image](https://user-images.githubusercontent.com/7662954/170938200-b9978a2e-26f5-4843-a5e5-5b6b42c5e3fd.png)

**Comment by SchmockLord** [comment]
Hi @henryxrl

I think I found the issue: it was 17 ports in the config and the port limit is 16. I removed HS12.

Can you replace your SSDT-UIAC-USB-Z490-ITXTB3.aml with this one and see if that helps.

[SSDT-UIAC-USB-Z490-ITXTB3.aml.zip](https://github.com/SchmockLord/Hackintosh-Intel-i9-10900k-AsRock-Z490-Phantom-ITX-TB3/files/8797530/SSDT-UIAC-USB-Z490-ITXTB3.aml.zip)

**Comment by henryxrl** [comment]
![image](https://user-images.githubusercontent.com/7662954/171031155-e7a72a19-932b-42d9-b030-2f2f7cf6d52f.png)

It seems all ports partially work - if I plug in a thumb drive, it'll show up in the system.

BUT, none of the USB ports works with other USB devices such as a speaker, keyboard, or mouse. This is extremely weird.

**Comment by henryxrl** [comment]
And Bluetooth doesn't work, although Wifi works. I replaced the original wifi card with the same one you recommended.

![image](https://user-images.githubusercontent.com/7662954/171051800-b01d15c1-157e-4e1c-b26b-1a6d0ec7423e.png)

**Comment by SchmockLord** [comment]
Please try this EFI. It is based on OC 0.8.1 and I have changed to USBMap.kext again.

But this config.plist uses iMac20,2. So enable iGPU.
[EFI_OC081.zip](https://github.com/SchmockLord/Hackintosh-Intel-i9-10900k-AsRock-Z490-Phantom-ITX-TB3/files/8800537/EFI_OC081.zip)

**Comment by henryxrl** [comment]
Thanks!! Now all USB ports fully work!!

Could you please still look into the Bluetooth issue that I have? That "Address: NULL" doesn't seem right to me...

**Comment by SchmockLord** [comment]
Great. Can you export the PCI section in Hackintool and share it?

So u replaced the onboard Intel by the Broadcom one? Are u sure it is a BCM94360NG?

**Comment by henryxrl** [comment]
Yes it is a BCM94360NG. 

![IMG_2809](https://user-images.githubusercontent.com/7662954/171056540-e8a47b28-9652-419c-8e2f-622be6754a46.jpeg)

Here's the PCI section:

![image](https://user-images.githubusercontent.com/7662954/171056586-c3db4349-ec7a-4add-ad39-6e88c99f046c.png)

**Comment by SchmockLord** [comment]
Ok. But is it working or not? Try with Airpods if you have. You can then also check if you have stable music playback.

**Comment by henryxrl** [comment]
No, the Bluetooth doesn't work. I can't turn it on - thus can't connect any devices like Airpods.

![image](https://user-images.githubusercontent.com/7662954/171057414-c6f4164a-6693-4c04-9f18-4e3ef767eb25.png)

**Comment by SchmockLord** [comment]
Try this. Added the BlueToolFixup.kext

https://github.com/acidanthera/BrcmPatchRAM#bluetoolfixupkext

And please send me screenshot of your Hackintool USB section.
[EFI_OC081_v2.zip](https://github.com/SchmockLord/Hackintosh-Intel-i9-10900k-AsRock-Z490-Phantom-ITX-TB3/files/8800718/EFI_OC081_v2.zip)

**Comment by henryxrl** [comment]
I tried the v2, but unfortunately, Bluetooth still doesn't work. The following is the screenshot of the USB section:

![image](https://user-images.githubusercontent.com/7662954/171079912-b4fbc85f-5a67-486e-ac91-59c2007ede9b.png)

BTW, the "Reset NVRAM" option no longer exists at boot time after using OC081. It seems AllowNVRAMReset is set to true and HideAuxiliary is set to false already. And I tried to press the space bar during the option selection screen but still no luck. Weird...

**Comment by SchmockLord** [comment]
Please press the broom and then refresh button in hackintool and send me the screenshot again.

**Comment by henryxrl** [comment]
Here you go:

![image](https://user-images.githubusercontent.com/7662954/171192122-57550065-c1bb-4ef6-b8bc-76565627057b.png)

![image](https://user-images.githubusercontent.com/7662954/171192160-2931e279-d3eb-402e-820d-2634a2dd9018.png)

**Comment by SchmockLord** [comment]
I think the issue is, that the Wifi-card is on a HSxx port which is not mapped and therefore doesn't get USB power to drive the Bluetooth. Can you try to disable the USB kext, restart and then screenshot again?

You have to find out on which HS port the Wifi card sits. I didn't document this for this board.

**Comment by henryxrl** [comment]
Hmm, how can I disable USB kext? Sorry, I'm new to this.

From this [url](https://github.com/Juan-VC/Hackintosh-macOS-Dell-G7-7588/issues/60), it seems the internal USB port for the NG wifi card is at HS07. (If you search for "HS07" you will find the place on the webpage)

BTW, I start to feel it might be that the wifi card is defective... I installed Windows 11 on the side. Even after I installed the driver for this wifi card (wifi driver + Bluetooth driver), there's still no working Bluetooth in Windows (wifi works in Windows).

**Comment by SchmockLord** [comment]
It is not always HS07. This varies from board to board. Most likely HS14 or HS12 or HS11. But all of these are already mapped. And if you say you don't have Bluetooth in Windows, then this indicates indeed, that you should try a different card.

**Comment by henryxrl** [comment]
Yeah... just that getting another one and installing it in takes a ton of work, sigh.

Oh, I just found another weird thing - the ethernet doesn't work. Everything shows up fine in the "System Information" and in hackintool but when I actually plug the cable in, it always shows no cable plugged. (The same cable works in Windows). Any suggestion on that?

And, also an issue I previously mentioned: "BTW, the "Reset NVRAM" option no longer exists at boot time after using OC081. It seems AllowNVRAMReset is set to true and HideAuxiliary is set to false already. And I tried to press the space bar during the option selection screen but still no luck. Weird..."

**Comment by SchmockLord** [comment]
Regarding Ethernet. Try to set it to manual detection and then 1Gbps.

**Comment by henryxrl** [comment]
I just noticed that the wifi card I got is FV-BCMM6-57167-65P-**055P**-B12 while everyone else online has FV-BCMM6-57167-65P-**598M**-B12. Maybe that's affecting the Bluetooth functionality. I'll buy another one and report back.

**Comment by henryxrl** [comment]
I confirm that it WAS indeed a hardware issue. Now that I've changed a card, the Bluetooth works flawlessly. (I used the EFI with OC 0.8.1 WITHOUT the BlueToolFixup.kext)

![image](https://user-images.githubusercontent.com/7662954/172277462-24a0fcb9-c5e0-4f1a-a3e3-24e46b98fa0b.png)
It changed from the previous "0x0001 Apple" to "0x4500 Broadcom Corp.". 

I do notice that the "FW Load" option under Bluetooth isn't checked. Is it a potential issue?
